install/kubernetes/cilium/templates/cilium-operator/deployment.yaml

```
@@ -423,5 +428,22 @@ spec:
                 path: local-etcd-client-ca.crt
           {{- end }}
       {{- end }}
+      {{- if and .Values.operator.prometheus.enabled .Values.operator.prometheus.tls.enabled }}
+      # To read the prometheus configuration
+      - name: prometheus-tls
+        projected:
+          # note: the leading zero means this number is in octal representation: do not remove it
+          defaultMode: 0400
+          sources:
+          - secret:
+              name: {{ .Values.operator.prometheus.tls.server.existingSecret | default "prometheus-server-certs" }}
+              items:
```

```suggestion
              name: {{ .Values.operator.prometheus.tls.server.existingSecret | default "prometheus-server-certs" }}
              optional: true
              items:
```
I suggest we don't require the secret to prevent blocking pod startup when the secret is not yet available (ex.: using job or cert-manager). This goes hand-in-hand with using the certloader mechanism and awaiting the promise.